src/paths.rs

```diff
@@ -20,3 +18,34 @@ pub struct Paths {
     /// Actual length the user said was maximum for images
     pub actual_max_viewable: usize,
 }
+
+/// Pushes images to vector (v) if the path (p) extension is a valid image extension
+pub fn push_image_path(v: &mut Vec<PathBuf>, p: PathBuf) {
+    if let Some(ext) = p.extension() {
+        if let Some(ext_str) = ext.to_str() {
+            let low = ext_str.to_string().to_lowercase();
+            if low == "jpg" || low == "jpeg" || low == "png" || low == "bmp" || low == "webp" {
+                v.push(p)
+            }
+        }
+    }
+}
+
+/// Constructs a normalized_path removing '.', parent_dir/'..'
+pub fn normalize_path(path: &PathBuf) -> PathBuf {
```

Comment:
@NickHackman 
What does `normalize_path` do (or do better) that `canonicalize_path` does not?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=472f3574408fdfbd2c507b00d2245ca5

Reply:
Nothing, I'm dumb and was reinventing the wheel...